infra/prover/index.ts

```diff
@@ -128,9 +137,9 @@ export = () => {
     }),
   });
 
-  brokerEcr.repository.arn.apply(arn => {
+  pulumi.all([brokerEcr.repository.arn, executionRole.name]).apply(([ecrRepoArn, executionRoleName]) => {
     new aws.iam.RolePolicy(`${serviceName}-ecs-execution-pol`, {
-      role: executionRole.id,
+      role: executionRoleName,
```

Comment:
you don't need the `.apply` function here, im pretty sure you can just:
```
role: executionRole.name
```
You're also not using `ecrRepoArn` that i can see.